### ISSUE TITLE
Issue with global_position while tween and pivot offset is set with certain values

### ISSUE THREAD

**Issue by CrankyUnicorn** [godotengine/godot#93716]
### Tested versions

Godot 4.2.1

### System information

Win 11 

### Issue description

Issue with tween weird behavior:  it keeps moving the square further and further away from the center of the screen

```gdscript
extends ColorRect
var tween:Tween

func _ready():
    self.pivot_offset = (self.size / 2)
    
    tween = create_tween().set_parallel(true)
    tween.tween_property(self, "rotation_degrees", 15.0, 0.6)
    # tween.tween_property(self, "scale", Vector2(1.05f, 1.05f), 0.6) # also with this

func _process(delta):
    self.global_position = self.global_position
```

Diminishing the pivot_offset to zero mitigates the issue 

### Steps to reproduce

single scene with squares colorRect with this script on it

```gdscript
extends ColorRect
var tween:Tween

func _ready():
    self.pivot_offset = (self.size / 2)
    
    tween = create_tween().set_parallel(true)
    tween.tween_property(self, "rotation_degrees", 15.0, 0.6)
    # tween.tween_property(self, "scale", Vector2(1.05f, 1.05f), 0.6) # also with this

func _process(delta):
    self.global_position = self.global_position
```

### Minimal reproduction project (MRP)

```gdscript
extends ColorRect
var tween:Tween

func _ready():
    self.pivot_offset = (self.size / 2)
    
    tween = create_tween().set_parallel(true)
    tween.tween_property(self, "rotation_degrees", 15.0, 0.6)
    # tween.tween_property(self, "scale", Vector2(1.05f, 1.05f), 0.6) # also with this

func _process(delta):
    self.global_position = self.global_position
```

**Comment by AwesomeAxolotl** [comment]
Can't reproduce in 4.3 beta1. Can reproduce in 4.2.2. So looks like it'll be fixed in 4.3.

**Comment by CrankyUnicorn** [comment]
> Can't reproduce in 4.3 beta1. Can reproduce in 4.2.2. So looks like it'll be fixed in 4.3.

true tested also but in the 4.3 beta2 there is a related issue now: with the same code try to scale only with scale:x as so

`tween_flip.tween_property(self, "scale:x", 1.0, 0.2)` 

Notice that the pivot should be and is based on the testing on the center of the object but on scaling:x (not on only scale, that is correct) is now on x=0,

**Comment by Chaosus** [comment]
> Can't reproduce in 4.3 beta1.

Thanks, indeed its not reproducable anymore. Closed.
